core/plannernotes.c

```diff
@@ -30,16 +30,60 @@ int diveplan_duration(struct diveplan *diveplan)
 	return (duration + 30) / 60;
 }
 
+struct icd_data { // This structure provides communication between function isobaric_counterdiffusion() and the calling software.
+	int dN2;      // The change in fraction (permille) of nitrogen during the change
+	int dHe;      // The change in fraction (permille) of helium during the change
+};
+
+/* Perform isobaric counterdiffusion calculations for gas changes in trimix dives.
+ * Here we use the rule-of-fifths where, during a change involving trimix gas, the increase in nitrogen
+ * should not exceed one fifth of the decrease in helium.
+ * Parameters: 1) Pointer to the dive structure.
+ *             2) pointers to two gas mixes, the gas being switched from and the gas being switched to.
+ *             3) a pointer to an icd_data structure.
+ * Output:     i) The icd_data stucture is filled with the delta_N2 and delta_He numbers (as permille).
+ *            ii) Function returns a boolean indicating an exceeding of the rule-of-fifths. False = no icd problem.
+ */
+bool isobaric_counterdiffusion(struct dive *dive, struct gasmix *oldgasmix, struct gasmix *newgasmix, struct icd_data *results)
```

Comment:
Now that I see it, struct dive is no longer needed, is it?

Reply:
Indeed. corrected. Kind regards, willem